base/REPL.jl

```diff
@@ -153,9 +153,9 @@ function print_response(errio::IO, val::ANY, bt, show_value::Bool, have_color::B
                 if val !== nothing && show_value
                     try
                         if specialdisplay === nothing
-                            display(val)
+                            eval(Main, Expr(:call, () -> display(val)))
```

Comment:
This is https://github.com/JuliaLang/julia/issues/6763. For example, QuoteNode still fails to maintain the egal identity of its contents and segfaults if it contains an invalid Expr such as:
```eval(Expr(:call, display, QuoteNode(Expr(:top))))```